api/download_from_s3.py

```diff
@@ -174,18 +178,48 @@ def get_relevant_html_keys_for_url(s3_client: client, bucket: str, url: str) ->
     return [o['Key'] for o in contents if o['Key'].startswith(url_without_https) and o['Key'].endswith('.html')]
 
 
-def get_relevant_png_keys_for_url(s3_client: client, bucket: str, url: str) -> list[str]:
+def get_relevant_png_keys_for_url_from_s3(s3_client: client, bucket: str, url: str) -> list[str]:
     """Returns a list of object keys from a given bucket, given a constraint."""
     if 'https' in url:
         url_without_https = url[8:]
     else:
         url_without_https = url
-
+    print(url)
     contents = s3_client.list_objects(Bucket=bucket)['Contents']
 
     return [o['Key'] for o in contents if o['Key'].startswith(url_without_https) and o['Key'].endswith('.png')]
 
 
+def retrieve_searched_for_pages(s3_client: client, input: str):
+    """Get the relevant pages that have been searched for."""
+
+    pages = []
+    keys = get_object_keys(s3_client, environ['S3_BUCKET'])
+    if keys is None:
+        return 'Empty Database!'
+
+    png_keys = filter_keys_by_type(keys, '.png')
+    relevant_keys = filter_keys_by_website(png_keys, input)
+    for relevant_key in relevant_keys:
+        display_key = relevant_key.split(
+            '/')[0] + '/' + relevant_key.split('/')[1]
+        pages.append(display_key)
+    return pages
```

Comment:
Perhaps this would be neater?
```suggestion
    keys = get_object_keys(s3_client, environ['S3_BUCKET'])
    if keys is None:
        return 'Empty Database!'

    png_keys = filter_keys_by_type(keys, '.png')
    relevant_keys = filter_keys_by_website(png_keys, input)
    
    return [relevant_key.split(
            '/')[0] + '/' + relevant_key.split('/')[1] for relevant_key in relevant_keys]
```